ldap/ldap.go

```diff
@@ -0,0 +1,11 @@
+// Package ldap provides LDAP-specific interfaces for service extensions.
+package ldap
+
+import "context"
+
+// Provider is the interface for LDAP operations.
+type Provider interface {
+	// BindDN retrieves the bind DN from the context. It returns an empty string
```

Comment:
It may be worth mentioning that the context should be retrieved from the [service extension API](https://github.com/strata-io/service-extension/blob/main/orchestrator/orchestrator.go#L53-L56).